lib/ansible/modules/cloud/vmware/vmware_drs_group.py

```diff
@@ -0,0 +1,600 @@
+        """
+        Function to get group by name
+        Args:
+            group_name: Name of group
+            cluster_obj: vim Cluster object
+
+        Returns: Group Object if found or None
+
+        """
+
+        if group_name is None:
+            group_name = self.__group_name
+
+        if cluster_obj is None:
+            cluster_obj = self.__cluster_obj
+
+        # Allow for group check even if dry run
+        if self.module.check_mode and cluster_obj is None:
+            return None
+
+        for group in cluster_obj.configurationEx.group:
+            if group.name == group_name:
+                return group
+
+        # No group found
+        return None
+
+    def __group_exists(self, group_name=None, cluster_obj=None):
+        """
+        Function to check if group already exists
+        Args:
+            group_name: Name of group
+            cluster_obj: vim Cluster object
+
+        Returns: Bool
+
+        """
+
+        if group_name is None:
+            group_name = self.__group_name
+
+        if cluster_obj is None:
+            cluster_obj = self.__cluster_obj
+
+        group = self.__get_group_by_name(group_name=group_name, cluster_obj=cluster_obj)
+
+        if group is None:
+            return False
+        else:
+            return True
+
+    def __populate_vm_host_list(self, group_name=None, cluster_obj=None, host_group=False):
+        """
+        Return all VM/Host names using given group name
+        Args:
+            group_name: group name
+            cluster_obj: Cluster managed object
+            host_group: True if we want only host name from group
+
+        Returns: List of VM/Host names belonging to given group object
+
+        """
+        obj_name_list = []
+
+        if group_name is None:
+            group_name = self.__group_name
+
+        if cluster_obj is None:
+            cluster_obj = self.__cluster_obj
+
+        if not all([group_name, cluster_obj]):
+            return obj_name_list
+
+        group = self.__get_group_by_name()
+
+        if not host_group and isinstance(group, vim.cluster.VmGroup):
+            obj_name_list = [vm.name for vm in group.vm]
+
+        elif host_group and isinstance(group, vim.cluster.HostGroup):
+            obj_name_list = [host.name for host in group.host]
+
+        return obj_name_list
+
+    def __check_if_vms_hosts_changed(self, group_name=None, cluster_obj=None, host_group=False):
+        """
+        Function to check if VMs/Hosts changed
+        Args:
+            group_name: Name of group
+            cluster_obj: vim Cluster object
+            host_group: True if we want to check hosts, else check vms
+
+        Returns: Bool
+
+        """
+
+        if group_name is None:
+            group_name = self.__group_name
+
+        if cluster_obj is None:
+            cluster_obj = self.__cluster_obj
+
+        list_a = self.__vm_list if host_group is False else self.__host_list
```

Comment:
```suggestion
        list_a = self.__host_list if host_group else self.__vm_list
```
Just a little suggestion to keep it easier/cleaner